parmed/amber/amberformat.py

```diff
@@ -490,36 +497,43 @@ def rdparm_slow(self, fname):
         fmtre = re.compile(r'%FORMAT *\((.+)\)')
         version = None
 
+        if isinstance(fname, string_types):
+            prm = genopen(fname, 'r')
+            own_handle = True
+        else:
+            prm = fname
+            own_handle = True
+            assert hasattr(prm, 'read'), 'must be a file object'
```

Comment:
Can you change the assert to a `raise TypeError('%s must be a file name or file-like object' % prm)`?